### PR TITLE
0.4 Codebase

### PR DESCRIPTION
**Pull request by AionJayT**
## Notice

It is not allowed to submit your PR to the master branch directly, please submit your PR to the master-pre-merge branch.

## Description
- Fork logic
- Monetary policy change
- Avm contract enable

Fixes Issue # .

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [x] New feature.
- [ ] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

-

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [ ] I have made appropriate changes to the corresponding documentation.
- [ ] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

File: modMcf/test/org/aion/mcf/config/CfgVmTest.java [hunk 1]
```diff
@@ -9,15 +9,6 @@
 import org.junit.Test;
 
 public class CfgVmTest {
```
Comment (AlexandraRoatis) [inline modMcf/test/org/aion/mcf/config/CfgVmTest.java]: this test can be removed entirely

File: modMcf/src/org/aion/mcf/config/CfgVm.java [hunk 2]
```diff
@@ -10,12 +10,6 @@
 
 /** Configuration section for VM */
 public class CfgVm {
```
Comment (AlexandraRoatis) [inline modMcf/src/org/aion/mcf/config/CfgVm.java]: this class is no longer necessary

File: modMcf/test/org/aion/mcf/config/CfgVmTest.java (hunk 1 above)
Comment (AionJayT) [reply modMcf/test/org/aion/mcf/config/CfgVmTest.java]: removed

File: modMcf/src/org/aion/mcf/config/CfgVm.java (hunk 2 above)
Comment (AionJayT) [reply modMcf/src/org/aion/mcf/config/CfgVm.java]: removed this class

**Pull request merged**